src/dock.c

```diff
@@ -251,12 +255,31 @@ void Create(TrayComponentType *cp)
 
    }
 
+   /* Create and assign dock background pixmap. */
+   cp->pixmap = JXCreatePixmap(display, rootWindow, cp->width, cp->height,
+                               rootDepth);
+   tp->buffer = cp->pixmap;
+   ClearTrayDrawable(cp);
+   XSetWindowBackgroundPixmap(display, cp->window, tp->buffer);
+
 }
 
 /** Resize a dock component. */
 void Resize(TrayComponentType *cp)
 {
+   DockType *tp = (DockType*)cp->object;
+
    JXResizeWindow(display, cp->window, cp->width, cp->height);
+
+   /* Recreate and assign dock background pixmap. */
+   if(tp->buffer != None) {
+      JXFreePixmap(display, tp->buffer);
+   }
+   cp->pixmap = JXCreatePixmap(display, rootWindow, cp->width, cp->height,
+                               rootDepth);
+   tp->buffer = cp->pixmap;
+   ClearTrayDrawable(cp);
+   XSetWindowBackgroundPixmap(display, cp->window, tp->buffer);
```

Comment:
Not that it's particularly important, but I think you meant to call `JXSetWindowBackgroundPixmap` here instead of `XSetWindowBackgroundPixmap`.